cli/smoke.go

```diff
@@ -228,3 +267,113 @@ func (c *SmokeTestEndpointCommand) checkEndpoints() error {
 	}
 	return catcher.Resolve()
 }
+
+// APIBuild represents part of a build from the REST API
+type APIBuild struct {
+	Tasks []string `json:"tasks"`
+}
+
+// APITask represents part of a task from the REST API
+type APITask struct {
+	Status string            `json:"status"`
+	Logs   map[string]string `json:"logs"`
+}
+
+func (c *SmokeTestEndpointCommand) checkTaskByCommit() error {
+	var builds []APIBuild
+	var build APIBuild
+	for i := 0; i <= 300; i++ {
+		// get task id
+		if i == 300 {
+			return errors.New("error getting builds for version")
+		}
+		time.Sleep(time.Second)
+		grip.Infof("checking for a build of %s (%d/300)", c.Commit, i+1)
+		resp, err := c.client.Get(urlPrefix + uiPort + "/rest/v2/versions/evergreen_" + c.Commit + "/builds")
+		if err != nil {
+			grip.Info(err)
+			continue
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			err = errors.Wrap(err, "error reading response body")
+			grip.Error(err)
+			return err
+		}
+		err = json.Unmarshal(body, &builds)
+		if err != nil {
+			err = json.Unmarshal(body, &build)
+			if err != nil {
+				return errors.Wrap(err, "error unmarshaling json")
+			}
+		}
+		if len(builds) == 0 {
+			builds = []APIBuild{build}
+		}
+		if len(builds[0].Tasks) == 0 {
+			grip.Info("no tasks found")
+			continue
+		}
+		break
+	}
+
+	var task APITask
+	for i := 0; i <= 300; i++ {
```

Comment:
why 300?

Reply:
5 minutes seemed plenty from testing this by hand. 1 minute was a little fast since you have to wait for runners to cycle. I can make it faster once it's running in Evergreen.